src/plugins/profiling/common/topn.ts

```diff
@@ -13,42 +13,74 @@ import {
 
 import { StackFrameMetadata } from './profiling';
 
-type TopNBucket = {
-  Value: string;
+export type TopNSample = {
+  Timestamp: number;
   Count: number;
+  Category: string;
 };
 
-type TopNBucketsByDate = {
-  TopN: Record<number, TopNBucket[]>;
+export type TopNSamples = {
```

Comment:
I think the original purpose was to create a union type for a response that either has a `Metadata` field (top N traces) or doesn't have the `Metadata` field (everything else). The type aliases were needed to satisfy the type system.

Since the client-server API is still in flux and a later refactor may require a different design, I switched to interfaces and removed the now-unnecessary type aliases and union type. We may still eventually need to create a response type to accomplish the original purpose, but for now it's not needed.